src/Simulation/Core/IQuantumMachineOutput.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// Interface to access the results of a program executed in a quantum machine.
+    /// </summary>
+    public interface IQuantumMachineOutput
+    {
+
+        /// <summary>
+        /// Gets an output probabilistically.
+        /// Calls to this method are not deterministic, each call can return a different output.
+        /// </summary>
+        object GetOutput();
+
+        /// <summary>
+        /// Gets the most frequent output.
+        /// If multiple outputs have the same frequency, it gets the first one in ascending order.
+        /// </summary>
+        object GetMostFrequentOutput();
```

Comment:
Similarly, this could also be obtained by using the `Histogram` property, so I am not sure that it should be a separate method.

Reply:
As with the GetOutput method, this is a convenience method. Someone can achieve the same result from the Histogram property but it requires a bit more code than just calling a method. It also allows the class that implements the IQuantumMachine interface to provide a consistent return value when there is ambiguity (e.g. when multiple outputs have the same frequency).

Reply:
We chatted offline and agreed that `GetOutput` and `GetMostFrequentOutput` could be moved to extension methods.